src/tests/unit/tests/electron/platform/android/setup/state-machine/state-machine.test.ts

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { Action } from 'common/flux/action';
+import { StateMachine } from 'electron/platform/android/setup/state-machine/state-machine';
+import { StateMachineSteps } from 'electron/platform/android/setup/state-machine/state-machine-steps';
+import { ExpectedCallType, It, Mock, MockBehavior, Times } from 'typemoq';
+
+type MartiniStepId = 'gin' | 'vermouth' | 'olives';
+
+class MartiniActions {
+    public shake = new Action<void>();
+    public stir = new Action<void>();
+    public garnish = new Action<string>();
+}
+
+type MartiniSteps = StateMachineSteps<MartiniStepId, MartiniActions>;
+type MartiniStepTransition = (stepId: MartiniStepId) => void;
```

Comment:
🍸 :) 

But also @RobGallo I've not seen the Martini pattern with state machines before. Do you know where I could read more about implementing this/look at other examples?